Update to RimWorld version 1.1

SaucyPigeon opened the pull request:
- Multi-version support for RimWorld versions 1.0 and 1.1.
- Separate project for each version, due to different Harmony and .NET versions.
- Some assembly references have been changed to package references for easier multi-version development.

SaucyPigeon committed:
Added package references

Easier versioning

SaucyPigeon committed:
Added 1.1 project, remove HugsLib package reference

For some reason, the HugsLib nuget package copies to output automatically even though it will give an error when you load the game...

SaucyPigeon committed:
Got rid of loose files

SaucyPigeon committed:
Updated About.xml, reverted debugging for 1.0 project

SaucyPigeon committed:
Merge pull request #1 from SaucyPigeon/development

Development

RangelReale commented:
I tried to run your pull request on my RimWorld install, that is already at 1.1, but got these errors, have any idea about it?
--------------------------------
ReflectionTypeLoadException getting types in assembly RR_PawnBadge-RW1.1: System.Reflection.ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
  at (wrapper managed-to-native) System.Reflection.Assembly.GetTypes(System.Reflection.Assembly,bool)
  at System.Reflection.Assembly.GetTypes () [0x00000] in <567df3e0919241ba98db88bec4c6696f>:0 
  at Verse.ModAssemblyHandler.AssemblyIsUsable (System.Reflection.Assembly asm) [0x00000] in <a12d1ce36500429788a966a010bef562>:0 

Loader exceptions:
   => System.TypeLoadException: Could not resolve type with token 01000022 (from typeref, class/assembly HugsLib.ModBase, HugsLib, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null)

Verse.Log:Error(String, Boolean)
Verse.ModAssemblyHandler:AssemblyIsUsable(Assembly)
Verse.ModAssemblyHandler:ReloadAll()
Verse.ModContentPack:ReloadContent()
Verse.LoadedModManager:LoadModContent()
Verse.LoadedModManager:LoadAllActiveMods()
Verse.PlayDataLoader:DoPlayLoad()
Verse.PlayDataLoader:LoadAllPlayData(Boolean)
Verse.<>c:<Start>b__6_1()
Verse.LongEventHandler:RunEventFromAnotherThread(Action)
Verse.<>c:<UpdateCurrentAsynchronousEvent>b__27_0()
System.Threading.ThreadHelper:ThreadStart_Context(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
System.Threading.ThreadHelper:ThreadStart()

An exception was caused by the HugsLibChecker assembly. Exception was: System.MissingMethodException: void Verse.LongEventHandler.QueueLongEvent(System.Action,string,bool,System.Action`1<System.Exception>)
  at HugsLibChecker.HugsLibChecker.RunAllChecks () [0x00056] in <17e0364003f64863ad8732cb22ef5516>:0 
Verse.Log:Error(String, Boolean)
HugsLibChecker.HugsLibChecker:RunAllChecks()
HugsLibChecker.HugsLibChecker:.ctor(ModContentPack)
System.Reflection.MonoCMethod:InternalInvoke(Object, Object[], Exception&)
System.Reflection.MonoCMethod:InternalInvoke(Object, Object[])
System.Reflection.MonoCMethod:DoInvoke(Object, BindingFlags, Binder, Object[], CultureInfo)
System.Reflection.MonoCMethod:Invoke(BindingFlags, Binder, Object[], CultureInfo)
System.RuntimeType:CreateInstanceImpl(BindingFlags, Binder, Object[], CultureInfo, Object[], StackCrawlMark&)
System.Activator:CreateInstance(Type, BindingFlags, Binder, Object[], CultureInfo, Object[])
System.Activator:CreateInstance(Type, Object[])
Verse.LoadedModManager:CreateModClasses()
Verse.LoadedModManager:LoadAllActiveMods()
Verse.PlayDataLoader:DoPlayLoad()
Verse.PlayDataLoader:LoadAllPlayData(Boolean)
Verse.<>c:<Start>b__6_1()
Verse.LongEventHandler:RunEventFromAnotherThread(Action)
Verse.<>c:<UpdateCurrentAsynchronousEvent>b__27_0()
System.Threading.ThreadHelper:ThreadStart_Context(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
System.Threading.ThreadHelper:ThreadStart()

SaucyPigeon commented:
I was able to replicate the error by not loading HugsLib. I'm going to commit a change to about.xml that says that this mod depends on it using the new 1.1 features.

SaucyPigeon committed:
About.xml includes HugsLib dependency.

This mod says that it depends on HugsLib, and also says that it needs to be loaded after it.

SaucyPigeon committed:
Update Harmony version to 2.0.0.7

SaucyPigeon committed:
Removed unneeded references for 1.0

SaucyPigeon committed:
Added scroll bar to badge menu

SaucyPigeon committed:
Merge pull request #2 from SaucyPigeon/development

Development

SaucyPigeon commented:
- Scroll bar to badge menu
- Harmony version update
